mwdb/resources/remotes.py

```diff
@@ -0,0 +1,459 @@
+
+    remote_url = app_config.get_key(f"remote:{remote_name}", "url")
+    api_key = app_config.get_key(f"remote:{remote_name}", "api_key")
+    return remote_url, api_key
+
+
+def map_remote_api_error(code):
+    if code == 200:
+        return None
+    elif code == 404:
+        raise NotFound("Remote object not found")
+    elif code == 403:
+        raise Forbidden("You are not permitted to perform this action on remote instance")
+    elif code == 409:
+        raise Conflict("Remote object already exists in remote instance and is different type")
+    else:
+        raise InternalServerError("Internal Server Error")
+
+
+class APiRemotePullResource(Resource):
+    session = requests.Session()
+    ObjectType = None
+    ItemResponseSchema = None
+
+    on_created = None
+    on_reuploaded = None
+
+    def create_pulled_object(self, item, is_new):
+        try:
+            db.session.commit()
+
+            if is_new:
+                hooks.on_created_object(item)
+                self.on_created(item)
+            else:
+                hooks.on_reuploaded_object(item)
+                self.on_reuploaded(item)
+        finally:
+            item.release_after_upload()
+
+        logger.info(f'{self.ObjectType.__name__} added', extra={
+            'dhash': item.dhash,
+            'is_new': is_new
+        })
+        schema = self.ItemResponseSchema()
+        return schema.dump(item)
+
+
+class APiRemoteFilePullResource(APiRemotePullResource):
+    ObjectType = File
+    ItemResponseSchema = FileItemResponseSchema
+
+    on_created = hooks.on_created_file
+    on_reuploaded = hooks.on_reuploaded_file
+
+    @requires_authorization
+    def post(self, remote_name, identifier):
+        """
+        ---
+        summary: Pulls file from remote to local instance
+        description: |
+            Pulls file from the remote instance to the local instance
+        security:
+            - bearerAuth: []
+        tags:
+            - api_remote
+        parameters:
+            - in: path
+              name: remote_name
+              description: name of remote
+              schema:
+                type: string
+            - in: path
+              name: identifier
+              description: Object identifier (SHA256/SHA512/SHA1/MD5)
+              schema:
+                type: string
+        responses:
+            200:
+                description: Information about uploaded config
+                content:
+                  application/json:
+                    schema: FileItemResponseSchema
+            404:
+                description: When the search name of the remote instance is not figured in the application config
+            409:
+                description: Object exists yet but has different type
+        """
+        remote_url, api_key = get_remote_api_data(remote_name)
+        response = self.session.request("GET", f"{remote_url}/api/file/{identifier}",
```

Comment:
Maybe it's better to use `session.get` directly?
```suggestion
        response = self.session.get(f"{remote_url}/api/file/{identifier}",
```